drivers/platform/mbed/mbed_i2c.cpp

```diff
@@ -0,0 +1,215 @@
+ */
+int32_t mbed_i2c_init(struct i2c_desc **desc,
+		      const struct i2c_init_param *param)
+{
+	mbed::I2C *i2c;		// pointer to new I2C instance
+	struct mbed_i2c_desc *mbed_i2c_desc;	// pointer to mbed i2c desc
+	struct i2c_desc *i2c_desc;
+
+	if (!desc || !param)
+		return -EINVAL;
+
+	/* Create an i2c descriptor object for the device */
+	i2c_desc = (struct i2c_desc *)calloc(1, sizeof(*i2c_desc));
+	if (!i2c_desc)
+		return -ENOMEM;
+
+	// Address passed in parameter shifted left by 1 to form
+	// 7-bit i2c slave address (7 MSBs) and the LSB acts as
+	// r/w bit during i2c read/write operations
+	i2c_desc->slave_address = ((param->slave_address) << 1);
+
+	/* Configure and instantiate I2C protocol */
+	i2c = new I2C(
+		(PinName)(((struct mbed_i2c_init_param *)param->extra)->i2c_sda_pin),
+		(PinName)(((struct mbed_i2c_init_param *)param->extra)->i2c_scl_pin));
+	if (!i2c)
+		goto err_i2c;
+
+	/* Create the i2c mbed descriptor object to store new i2c instance */
+	mbed_i2c_desc = (struct mbed_i2c_desc *)calloc(1, sizeof(*mbed_i2c_desc));
+	if (!mbed_i2c_desc)
+		goto err_mbed_i2c_desc;
+
+	mbed_i2c_desc->i2c_port = i2c;
+	i2c_desc->extra = mbed_i2c_desc;
+
+	*desc = i2c_desc;
+
+	return SUCCESS;
+
+err_mbed_i2c_desc:
+	free(i2c);
+err_i2c:
+	free(i2c_desc);
+
+	return -ENOMEM;
+}
+
+/**
+ * @brief Free the resources allocated by mbed_i2c_init().
+ * @param desc[in] - The I2C descriptor.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t mbed_i2c_remove(struct i2c_desc *desc)
+{
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	/* Free the I2C port object */
+	if ((I2C *)(((mbed_i2c_desc *)(desc->extra))->i2c_port))
+		delete((I2C *)(((mbed_i2c_desc *)(desc->extra))->i2c_port));
+
+	/* Free the I2C descriptor objects */
+	free(desc->extra);
+	free(desc);
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Write data to a slave device.
+ * @param desc[in] - The I2C descriptor.
+ * @param data[in] - Buffer that stores the transmission data.
+ * @param bytes_number[in] - Number of bytes to write.
+ * @param stop_bit[in] - Stop condition control.
+ *                   Example: 0 - A stop condition will not be generated;
+ *                            1 - A stop condition will be generated.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t mbed_i2c_write(struct i2c_desc *desc,
+		       uint8_t *data,
+		       uint8_t bytes_number,
+		       uint8_t stop_bit)
+{
+	mbed::I2C *i2c;
+
+	if (!desc  || !desc->extra || !data)
+		return -EINVAL;
+
+	i2c = (I2C *)(((mbed_i2c_desc *)(desc->extra))->i2c_port);
+
+	/**
+	    The MBED I2C API is reversed for parameter 4
+	    Instead of stop_bit - it has
+	    @param repeated   - Repeated start, true - don't send stop at end default value is false.
+	    Inverting here to keep the no-OS/platform_drivers API
+	 */
+	if (!(i2c->write(desc->slave_address, (char *)data, bytes_number, !stop_bit))) {
```

Comment:
nit: no need for brackets in this `if ... else` statement. actually you can also get rid of `else`:
```
if ((i2c->write(desc->slave_address, (char *)data, bytes_number, !stop_bit)))
     return -EINVAL;

return SUCCESS;
```

Reply:
I have updated this code as per review comment.